### PR TITLE
Return expect to verify neo-cli

### PR DESCRIPTION
**Pull request by vncoelho**
Restoring the structure from https://github.com/neo-project/neo-node/tree/master/.github/workflows

part https://github.com/neo-project/neo/issues/3256

### CODE REVIEW

**Review by Jim8y — APPROVED** [review]

**Review by cschuchardt88 — CHANGES_REQUESTED** [review]
Please put scripts and other development files in `/.neo` folder of repo. You can create directories or files in there. That's what it for.

**Comment by Jim8y** [comment]
> Please put scripts and other development files in `/.neo` folder of repo. You can create directories or files in there. That's what it for.

So we can run the test locally?

**Comment by cschuchardt88** [comment]
> So we can run the test locally?
Yes, organization and for `csproj` files or `props` can use if needed.

**Review by Jim8y — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -24,11 +24,30 @@ jobs:
...
     - name: Build (Neo.CLI)
-      run: |
-        dotnet build ./src/Neo.CLI \
-        --output ./out/Neo.CLI
-
+      run: dotnet build ./src/Neo.CLI --output ./out/Neo.CLI
+        
+    - name: Install dependencies
+      run: |
+        sudo apt-get install libleveldb-dev expect
```
Comment (shargon) [inline .github/workflows/main.yml:45]: leveldb is already installed?

Comment (vncoelho) [reply .github/workflows/main.yml]: It was being installed below. I moved everything up and ensured dependency between tasks.

**Comment by vncoelho** [comment]
> Please put scripts and other development files in `/.neo` folder of repo. You can create directories or files in there. That's what it for.

.neo is not appropriate, a hidden folder.

I script, for now, a Script folder inside Neo.CLI itself.

File: .github/workflows/main.yml (hunk 1 above)
Comment (vncoelho) [reply .github/workflows/main.yml]: In fact,I will need to revert because each task need their dep

**Review by Jim8y — APPROVED** [review]

File: src/Neo.CLI/Scripts/test-neo-cli.expect
Comment (cschuchardt88) [inline src/Neo.CLI/Scripts/test-neo-cli.expect:1]: We don't scripts included the `csproj` files. Just create a folder `/scripts/Neo.CLI` and put there. If doesn't make sense to put in with source files.

**Review by cschuchardt88 — CHANGES_REQUESTED** [review]

File: src/Neo.CLI/Scripts/test-neo-cli.expect
Comment (Jim8y) [reply src/Neo.CLI/Scripts/test-neo-cli.expect:1]: You can fix it in another pr. If @vncoelho 's purpose is bring this back to the project. Please let him just bring it back.

Comment (shargon) [reply src/Neo.CLI/Scripts/test-neo-cli.expect:1]: I'm agree with do it organized, why we need to create one pr for one thing and other for move it? review is for this, the solution is not create a new pr.

Comment (Jim8y) [reply src/Neo.CLI/Scripts/test-neo-cli.expect:1]: > I'm agree with do it organized, why we need to create one pr for one thing and other for move it? review is for this, the solution is not create a new pr.

The problem is this pr from @vncoelho is to bring expert back, moving it somewhere is a request from @cschuchardt88 , which is not what @vncoelho wanted to do..... Why would you require/enforce @vncoelho to do a thing that is out of the scope of his pr? I am not against of making it organized, but what if @vncoelho just not willing to apply his suggestion? what we do with this pr then? close it and open another one?

Comment (vncoelho) [reply src/Neo.CLI/Scripts/test-neo-cli.expect:1]: His suggestion is good anyway, I will move to the main folder.

Comment (vncoelho) [reply src/Neo.CLI/Scripts/test-neo-cli.expect:1]: The point here is that Cris is using the Requested Changes too much.

I think that this is more useful when you are really blocking something for a good reason. 
These uses like this are not necessary.

Comment (vncoelho) [reply src/Neo.CLI/Scripts/test-neo-cli.expect:1]: > > I'm agree with do it organized, why we need to create one pr for one thing and other for move it? review is for this, the solution is not create a new pr.
> 
> The problem is this pr from @vncoelho is to bring expert back, moving it somewhere is a request from @cschuchardt88 , which is not what @vncoelho wanted to do..... Why would you require/enforce @vncoelho to do a thing that is out of the scope of his pr? I am not against of making it organized, but what if @vncoelho just not willing to apply his suggestion? what we do with this pr then? close it and open another one?
> 
> So my point is very simple, one pr for only one task, if you agree that we should bring expert back, and @vncoelho 's pr has fullfilled this purpose, we should be fine with this pr. You can have suggestions, but that should not be a reason of enforcing unless there is a real problem in the pr.
> 
> From our past working experience, i believe having small prs (real pr that can fix problem without introducing new problem), is the only way we can make it forward.
> 
> BUT, its just my personal suggestion.

I completely agree with you. This should be a common sense for most of the PRs.
Stop trying to block PRs for nonsense things that are not the goal of the PRs themselves.

**Review by Jim8y — APPROVED** [review]

**Review by shargon — APPROVED** [review]

File: src/Neo.CLI/Scripts/test-neo-cli.expect
Comment (cschuchardt88) [reply src/Neo.CLI/Scripts/test-neo-cli.expect:1]: > > I'm agree with do it organized, why we need to create one pr for one thing and other for move it? review is for this, the solution is not create a new pr.
> 
> The problem is this pr from @vncoelho is to bring expert back, moving it somewhere is a request from @cschuchardt88 , which is not what @vncoelho wanted to do..... Why would you require/enforce @vncoelho to do a thing that is out of the scope of his pr? I am not against of making it organized, but what if @vncoelho just not willing to apply his suggestion? what we do with this pr then? close it and open another one?
> 
> So my point is very simple, one pr for only one task, if you agree that we should bring expert back, and @vncoelho 's pr has fullfilled this purpose, we should be fine with this pr. You can have suggestions, but that should not be a reason of enforcing unless there is a real problem in the pr.
> 
> From our past working experience, i believe having small prs (real pr that can fix problem without introducing new problem), is the only way we can make it forward.
> 
> BUT, its just my personal suggestion.



> Something I would like to highlighted is that @cschuchardt88 is using the Requested Changes too much. I think that this is more useful when you are really blocking something for a good reason. These uses like this are not necessary and just reduce the criteria we will often look to these requests.

Than you should be creating a `Issue` and assigning someone to that task. I can't keep being the repo cleaner upper. Put some thought into it, that's all. Just remember no one can do it the right way but yourself. No copy/paste and hope.

**Pull request merged**